### DISCUSSION TITLE
3rd party authentication

### DISCUSSION THREAD

**Discussion by juesor** [peering-manager/peering-manager#664] [Ideas]
I leverage an external auth in nautobot for Azure AD.

Is this possible in peering-manager?

This is what i currently use in nautobotconfiguration.py and was hoping /opt/peeringmanager/peering_manager/configuration.py could do the same 

`  
# Global 3rd-party authentication settings
EXTERNAL_AUTH_DEFAULT_GROUPS = []
EXTERNAL_AUTH_DEFAULT_PERMISSIONS = {}

# 3rd-party authentication - Azure AD
AUTHENTICATION_BACKENDS = [
    "social_core.backends.azuread_tenant.AzureADTenantOAuth2",
    "nautobot.core.authentication.ObjectPermissionBackend",
]

SOCIAL_AUTH_AZUREAD_TENANT_OAUTH2_KEY = 'my key'
SOCIAL_AUTH_AZUREAD_TENANT_OAUTH2_SECRET = 'my secret'
SOCIAL_AUTH_AZUREAD_TENANT_OAUTH2_TENANT_ID = 'my tenant id'
SOCIAL_AUTH_AZUREAD_LOGIN_REDIRECT_URL = 'https://nautobot.abc.xyz/complete/azuread-tenant-oauth2/'
`

[[`](https://peering-manager.readthedocs.io/en/stable/setup/6-ldap/)](url)

Yes I've looked at the documentation for LDAP but i'd like to use Azure tenant auth.

**Comment by juesor** [comment]
Any suggesting?  Has anyone else done this? 

Anything?

**Reply by gmazoyer** [reply]
Social auth backend is not implemented as of today.

**Reply by juesor** [reply]
Is this on the release track in the future?

**Reply by gmazoyer** [reply]
It's not on the release track right now but it can be added. We should rework the whole authentication framework while we are at it.
